org-roam.el

```diff
@@ -821,6 +851,321 @@ for Org-ref cite links."
+  (interactive "P")
+  (when (org-roam--org-roam-file-p)
+    (org-element-map (org-element-parse-buffer) 'link
+      (lambda (link)
+        (let* ((type (org-element-property :type link))
+               (path (org-element-property :path link))
+               (is-title (if (string= type "roam") t nil))
+               (roam-file (if is-title (org-roam--get-file-from-title path) t)))
+          (unless roam-file
+            (org-roam--auto-create-file path manual)))))))
+
+(defun org-roam--current-buffer-roam-link-titles ()
+  "Return a list of unique roam-link titles in the current buffer."
+  (->> (org-element-map (org-element-parse-buffer) 'link
+         (lambda (link)
+           (let* ((type (org-element-property :type link))
+                  (path (org-element-property :path link))
+                  res)
+             (when (string= type "roam")
+               (cons path res)))))
+       (-flatten)
+       (-distinct)))
+
+(defun org-roam--roam-link-completion (&optional _arg)
+  "Completion for roam-links in `org-mode'.
+ARG is optional prefix supplied through `org-mode'"
+  (let ((completions (--> (org-roam--get-title-path-completions)))
+        (in-buffer-completions (--> (org-roam--current-buffer-roam-link-titles))))
+    (format "roam:%s" (completing-read "Roam note: "
+                                       (-union (map-keys completions) in-buffer-completions)))))
+
+(defun org-roam--roam-link-message ()
+  "Send roam-link status message to minibuffer."
+  (when (string= major-mode "org-mode")
+    (let ((elem (org-element-context)))
+      (when (and (eq (car elem) 'link)
+                 (string= "roam" (org-element-property :type elem)))
+        (save-match-data
+          (let* ((title (org-element-property :path elem))
+                 (abs-file-path (org-roam--get-file-from-title title))
+                 (rel-file-path
+                  (when abs-file-path
+                    (string-match (format "%s\\(.*\\)" org-roam-directory) abs-file-path)))
+                 (file-path (when rel-file-path
+                              (concat "~org-roam-dir~" (match-string 1 abs-file-path))))
+                 (raw-link (org-element-property :raw-link elem)))
+            (if file-path
+                (message "file: %s → %s" file-path raw-link)
+              (message "No file found in db → %s" raw-link))))))))
+
+
+(defun org-roam-show-link-messages ()
+  "Enable minibuffer status message for roam-links.
+Follows example of `org-ref' and displays on idle timer."
+  (interactive)
+  (or org-roam-link-message-timer
+      (setq org-roam-link-message-timer
+            (run-with-idle-timer 0.5 t #'org-roam--roam-link-message))))
+
+(defun org-roam-cancel-link-messages ()
+  "Disable minibuffer status message for roam-links."
+  (interactive)
+  (cancel-timer org-roam-link-message-timer)
+  (setq org-roam-link-message-timer nil))
+
+;; Turn roam-link status message on by default
+(org-roam-show-link-messages)
```

Comment:
I'm not sure we need this timer machinery here. We also have a variable `org-roam-verbose`, which we should use instead to control whether the messages are shown.

Reply:
#593

Reply:
The timer is lifted straight from org-ref and how it persistently shows the bibtex entry in the minibuffer on a cite-link (or a note that is doesn't exist). Without it, the link message disappears from the minibuffer fairly quickly. It seemed desirable to have it be persistent while on a link if you were using the message to resolve a filename, or a note title if you added a link description.

If we do remove the timer I would suggest changing/removing the link message functionality also.

I'll look into `org-roam-verbose`!

Reply:
The timer is now started on `org-roam-mode` activation when `org-roam-verbose` is `t`.

Reply:
Updated again to what I think is more along the lines of what you meant with `org-roam-verbose`

Reply:
ah I see, I thought it was a simple message, in which case it's fine, but also good to probably disable that if `org-roam-verbose` is nil.